### PR TITLE
[analyze] Add Analyzer for Opsgenie

### PR DESCRIPTION
**Pull request by abmussani**
### Description:
Implemented analyzer interface for opsgenie.

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/analyzer/analyzers/opsgenie/opsgenie.go [hunk 1]
```diff
@@ -14,8 +14,79 @@ import (
...
+		result.UnboundedResources = append(result.UnboundedResources, analyzers.Resource{
+			Name:               user.FullName,
+			FullyQualifiedName: user.FullName,
+			Type:               "user",
+			Metadata: map[string]any{
+				"username": user.Username,
+				"role":     user.Role.Name,
+			},
+		})
+	}
```
Comment (mcastorina) [inline pkg/analyzer/analyzers/opsgenie/opsgenie.go:85]: This is appending to `result.UnboundedResources`, which was allocated as a slice of non-zero length above: `make([]analyzers.Resource, len(info.Users))`.

Either change the assignment here to use the index:
```go
for i, user := range info.Users {
    result.UnboundedResources[i] = analyzers.Resource{
        // ...
    }
}
```

or change the initial assignment to allocate capacity:
```go
make([]analyzers.Resource, 0, len(info.Users))
```

**Review by mcastorina — CHANGES_REQUESTED** [review]
This is also missing the enumerated permissions / generated types.

File: pkg/analyzer/analyzers/opsgenie/opsgenie.go (hunk 1 above)
Comment (abmussani) [reply pkg/analyzer/analyzers/opsgenie/opsgenie.go]: Nice catch 🙇

**Comment by abmussani** [comment]
> This is also missing the enumerated permissions / generated types.

Add the enumerated permissions in the updates.

**Review by mcastorina — APPROVED** [review]

**Review by mcastorina — APPROVED** [review]

**Pull request merged**